app/(interview)/interview/new/page.tsx

```diff
@@ -3,57 +3,71 @@
-import { prisma } from '~/utils/db';
+import { trpc } from '~/app/_trpc/server';
+import { faker } from '@faker-js/faker';
 
-const createInterview = async (user: User, protocolId: string) => {
-  if (!user || !user.id) {
-    throw new Error('No user provided');
-  }
+export const dynamic = 'force-dynamic';
 
-  if (!protocolId) {
-    throw new Error('No protocol ID provided');
+export default async function Page({
+  searchParams,
+}: {
+  searchParams: {
+    identifier?: string;
+  };
+}) {
+  // check if active protocol exists
+  const activeProtocol = await trpc.protocol.getActive.query();
+  if (!activeProtocol) {
+    return (
+      <div className="flex h-screen items-center justify-center">
+        <div className="text-center">
+          <div className="mb-2 text-2xl font-bold">No Active Protocol</div>
+          <p className="max-w-md text-lg">
+            There is no active protocol for this account. Researchers may
+            optionally mark an uploaded protocol as active from the{' '}
+            <a href="/dashboard/protocols">protocols dashboard</a>.
+          </p>
+        </div>
+      </div>
+    );
   }
-
-  // eslint-disable-next-line local-rules/require-data-mapper
-  const interview = await prisma.interview.create({
-    data: {
-      user: {
-        connect: {
-          id: user.id,
-          // email: user.email,
-        },
+  // check if anonymous recruitment is enabled
+  const allowAnonymousRecruitment =
+    await trpc.metadata.get.allowAnonymousRecruitment.query(undefined, {
+      context: {
+        revalidate: 0,
       },
-      network: '',
-      protocol: {
-        connect: {
-          id: protocolId,
-        },
-      },
-    },
-  });
+    });
 
-  return interview;
-};
+  if (!allowAnonymousRecruitment) {
+    return (
+      <div className="flex h-screen items-center justify-center">
+        <div className="text-center">
+          <div className="mb-2 text-2xl font-bold">
+            Anonymous Recruitment Disabled
+          </div>
+          <p className="max-w-md text-lg">
+            Anonymous recruitment is disabled for this study. Reserachers may
+            optionally enable anonymous recruitment from the{' '}
+            <a href="/dashboard">main dashboard</a>.
+          </p>
+        </div>
+      </div>
+    );
+  }
 
-export default async function Page({ params, searchParams }) {
-  // Get the protocol ID from the search params
-  const { protocol } = searchParams;
+  // Anonymous recruitment is enabled
 
-  // Check we have a currently logged in user. Eventually, we want
-  // to optionally create a new user automatically as part of the
-  // onboarding flow.
-  const session = false;
-  console.error('UPDATE ME');
+  // Create a new interview
+  const identifier = searchParams.identifier || faker.string.uuid();
+  const { createdInterview } = await trpc.interview.create.mutate(identifier);
 
-  if (!session) {
-    redirect('/');
+  if (!createdInterview) {
```

Comment:
I think we need to handle the case that the identifier already exists. It seems like we might need to separate out the case where the identifier is provided in the searchParams and where it is generated. 

For situations where it is provided, we need to show a message that the identifier is in use. You should be able to update the return type of the mutation to have a property that can indicate this.

Reply:
Just while I think about it, there are other cases too: identifier isn't provided, identifier isn't valid (too long, short).

Reply:
resolved [ec1f4f3](https://github.com/complexdatacollective/Fresco/pull/20/commits/ec1f4f378d47d5b2b104a4c2d69fbbdafc55c6b8)